pkg/wolfssl/sock_tls/sock_tls.h

```diff
@@ -0,0 +1,339 @@
+ *                      There are several options for selecting the desired protocol.
+ *                      wolfSSL currently supports DTLS 1.0, and DTLS 1.2. Each of these
+ *                      protocols have a corresponding function that can be used as last
+ *                      argument to `sock_dtls_create`. The possible client and server
+ *                      protocol options are: `wolfDTLSv1_client_method()`, `wolfDTLSv1_server_method()`,
+ *                      `wolfDTLSv1_2_client_method()` and `wolfDTLSv1_server_method`.
+ *
+ * @return  0 on success.
+ * @return  -EADDRINUSE, if `local != NULL` and @p local is already used
+ *          elsewhere or if `local->port == 0` but the pool of ephemeral ports
+ *          is depleted
+ * @return  -EAFNOSUPPORT, if `local != NULL` or `remote != NULL` and
+ *          sock_udp_ep_t::family of @p local or @p remote is not supported.
+ * @return  -EINVAL, if sock_udp_ep_t::addr of @p remote is an invalid address.
+ * @return  -EINVAL, if sock_udp_ep_t::netif of @p local or @p remote are not a
+ *          valid interfaces or contradict each other (i.e.
+ *          `(local->netif != remote->netif) &&
+ *          ((local->netif != SOCK_ADDR_ANY_NETIF) ||
+ *          (remote->netif != SOCK_ADDR_ANY_NETIF))` if neither is `NULL`).
+ * @return  -ENOMEM, if not enough resources can be provided for `sock` to be
+ *          created, 0 upon success.
+ */
+int sock_dtls_create(sock_tls_t *sock, const sock_udp_ep_t *local, const sock_udp_ep_t *remote, uint16_t flags, WOLFSSL_METHOD *method);
+
+/**
+ * @brief   Sets the endpoint address for the remote DTLS peer associated to this sock object
+ *
+ * @pre `(sk != NULL)`
+ * @pre `(addr != NULL)`
+ *
+ * @param[in] sock     The resulting sock object.
+ * @param[in] addr      Remote end point for the DTLS session.
+ *
+ */
+void sock_dtls_set_endpoint(sock_tls_t *sk, const sock_udp_ep_t *addr);
+
+/**
+ * @brief   Creates a new DTLS session from an existing `sock_tls_t` object.
+ *
+ * @pre `(sk != NULL)`
+ *
+ * @param[in] sock      The sock object previously created using @ref sock_dtls_create
+ *
+ * @return  0 on success.
+ * @return  -EINVAL, if @sock is null or the SSL context is not initialized yet.
+ * @return  -ENOMEM, if not enough resources can be provided for the session to be
+ *          created
+ */
+int sock_dtls_session_create(sock_tls_t *sk);
+
+/**
+ * @brief   Destroys the DTLS session associated to the sock object.
+ *
+ * @pre `(sk != NULL)`
+ *
+ * @param[in] sk      The sock object containing the session to destroy.
+ *
+ */
+void sock_dtls_session_destroy(sock_tls_t *sk);
+
+/**
+ * @brief   Closes the socket connection associated to the DTLS session.
+ *
+ * @pre `(sk != NULL)`
+ *
+ * @param[in] sk      The sock object containing the UDP socket to close.
+ *
+ */
+void sock_dtls_close(sock_tls_t *sk);
+
+
+#ifdef MODULE_SOCK_TCP
+/* TODO */
```

Comment:
Shouldn't this trigger some kind of build error, as its currently not supported? Maybe add a little bit more information.on why this is not available.